### PR TITLE
Gf 1163 statusflow

### PR DESCRIPTION
**Pull request by BenKDishman**
Notes 
a) this should go in with corresponding business layer commit
b) this should wait for QA to accept before merge as it will break their current tests

Questions
1) I have added a comment to the applicationHandler for the unexpected error where there is no section corresponding to the request.  Do we want a generic error handler at some point, or have I missed the point ?  Presumably need something a) in principle and b) to lose compiler warnings

2) I've got a couple of redirects in the ActionHandler when moving from edit view to preview.  Is there a better way to do this (maybe move preview rendering to the action handler itself and inject the actionHandler to the previewController? ) 

Approvals
I'd say 2 approvals but 1 confident approval (more work required) is probably good enough.

### CODE REVIEW

File: src/main/twirl/views/sectionPreview.scala.html [hunk 1]
```diff
@@ -27,10 +29,28 @@ <h1 class="heading-xlarge">
 
             @fields.map(_.renderPreview(answers))
 
-            <div class="rifs-form-buttons">
-                <a class="button secondary" href="@controllers.routes.ApplicationController.show(app.id)">Close</a>
-                <a class="flat" href="@controllers.routes.ApplicationController.showSectionForm(app.id, formSection.sectionNumber)">Edit this page</a>
-            </div>
+            @if(editButton.isDefined){
+                <p></p>
```
Comment (jon-shipley) [inline src/main/twirl/views/sectionPreview.scala.html]: Is this empty <p> tag just to increase space?

File: src/main/twirl/views/sectionPreview.scala.html [hunk 2]
```diff
@@ -27,10 +29,28 @@ <h1 class="heading-xlarge">
...
+                        action=@editButton
+                        novalidate="novalidate">
+                        <a class="button secondary" href=@closeButton>Close</a>
+                        <input type="submit" class="button--link-style" value="Edit this page"/>
+                    </form>
+                </div>
+            }
+
+            @if(editButton.isEmpty){
+                <p></p>
```
Comment (jon-shipley) [inline src/main/twirl/views/sectionPreview.scala.html]: And another empty <p> tag.

File: src/main/scala/controllers/ApplicationController.scala [hunk 3]
```diff
@@ -32,19 +33,51 @@ class ApplicationController @Inject()(actionHandler: ActionHandler, applications
...
-        applications.getSection(id, sectionNumber).flatMap { section =>
+      case Some(fields) => {
+        applications.getSection(id, sectionNumber).flatMap { section => {
           val hints = section.map(s => hinting(s.answers, checksFor(sectionNumber))).getOrElse(List())
           actionHandler.renderSectionForm(id, sectionNumber, section, questionsFor(sectionNumber), fields, noErrors, hints)
-        }
+        }}
+      }
+      // Do we want a generic error handler for these kind of unlikely eventualities (+ avoid compiler warnings)
+      case None => ???
```
Comment (DougC) [inline src/main/scala/controllers/ApplicationController.scala]: Need to fix this. Suggest returning a 404 response - fieldsFor now returns values for all the known sections, so a None would represent a non-existent section number, so NotFound seems reasonable.

File: src/main/twirl/views/sectionPreview.scala.html (hunk 1 above)
Comment (BenKDishman) [inline src/main/twirl/views/sectionPreview.scala.html]: yes both are removed now (I was forgetting that the page was already productionised which means my fiddling is probably a little irritating!)

**Pull request merged**